questionnaires/tests.py

```diff
@@ -1023,3 +1023,30 @@ def test_csv_export_with_modified_questions(self):
             f'{self.user_submission_02.id},Survey 01,2022-08-31 23:00:00+00:00,question_01,c2\r\n'
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(byte_response.decode(), expected_response)
+
+
+class PollTest(TestCase):
+    def setUp(self):
+        Site.objects.all().delete()
+        self.site = SiteFactory(site_name='IoGT', port=8000, is_default_site=True)
+        self.home_page = self.site.root_page
+        self.poll = PollFactory(parent=self.home_page,)
+
+    def test_multi_select_option_results_for_checkboxes(self):
+        poll_question = PollFormFieldFactory(page=self.poll, field_type='checkboxes', choices='c1|c2|c3')
+
+        response = self.client.post(self.poll.url, {poll_question.clean_name: ('c1', 'c2')})
+        results = response.context_data['results']
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(results), 3)
+        self.assertEqual(results[0][0], 'c1')
+        self.assertEqual(results[1][0], 'c2')
+        self.assertEqual(results[2][0], 'c3')
+        self.assertEqual(results[0][1], 100.0)
+        self.assertEqual(results[1][1], 100.0)
+        self.assertEqual(results[2][1], 0.0)
+        self.assertTrue(results[0][2])
+        self.assertTrue(results[1][2])
+        self.assertFalse(results[2][2])
```

Comment:
I find it awkward that `results` is a list of tuples rather than say, a list of dicts, which I think might be easier to work with. I won't ask for this to be changed now though.

Reply:
Agreed, but I believe it's better to keep the PR relevant and to the point so it's not too difficult to review. Maybe we can add it to the backlog.